UPGRADING

```diff
@@ -734,6 +734,21 @@ PHP 8.0 UPGRADE NOTES
 
 - CURL:
   . The CURL extension now requires at least libcurl 7.29.0.
+  . curl_init() will now return a Curl object rather than a resource.
+    Return value checks using is_resource() should be replaced with
+    checks for `false`. The curl_close() function no longer has an effect,
+    instead the Curl instance is automatically destroyed if it is no longer
+    referenced.
+  . curl_multi_init() will now return a CurlMulti object rather than a resource.
+    Return value checks using is_resource() should be replaced with
+    checks for `false`. The curl_multi_close() function no longer has an effect,
+    instead the CurlMulti instance is automatically destroyed if it is no longer
+    referenced.
+  . curl_share_init() will now return a CurlShare object rather than a resource.
+    Return value checks using is_resource() should be replaced with
+    checks for `false`. The curl_share_close() function no longer has an effect,
+    instead the CurlShare instance is automatically destroyed if it is no longer
+    referenced.
```

Comment:
We should think a bit about class names... This introduces Curl, CurlMulti, CurlShare. Are these good?

I'd possibly go for something like `CurlHandle`.

Reply:
`CurlHandle` sounds ok. What do you think about a `CurlClient`? (I think this is my preference so far)

I'm uncertain about the other classes. `CurlMultiHandle`/`CurlMultiClient`? `CurlShareHandle`/`CurlShareClient`?

Reply:
Not a fan of CurlClient. To me that would imply that you can use the same CurlClient to fire off multiple requests, while here it only represents a single request (I guess using it for multiple is possible, but pretty weird).

Reply:
You won :) Since It's not a stateless class, you are right, that the handle name will be better for sure. Also, my friend suggested this as well.

Reply:
The UPGRADING notes will need to be updated as well.